diff --git a/dist/esm/server/lib/router-utils/proxy-request.js b/dist/esm/server/lib/router-utils/proxy-request.js
index 4a86de1386e7459b855dc4c461704143578042c6..3629ca428cc40e80458c35b8cca494b31f1ab9c2 100644
--- a/dist/esm/server/lib/router-utils/proxy-request.js
+++ b/dist/esm/server/lib/router-utils/proxy-request.js
@@ -2,17 +2,18 @@ import url from 'url';
 import { stringifyQuery } from '../../server-route-utils';
 import { Duplex } from 'stream';
 import { DetachedPromise } from '../../../lib/detached-promise';
+import HttpProxy from 'next/dist/compiled/http-proxy';
 export async function proxyRequest(req, res, parsedUrl, upgradeHead, reqBody, proxyTimeout) {
     const { query } = parsedUrl;
     delete parsedUrl.query;
     parsedUrl.search = stringifyQuery(req, query);
     const target = url.format(parsedUrl);
-    const HttpProxy = require('next/dist/compiled/http-proxy');
     const proxy = new HttpProxy({
         target,
         changeOrigin: true,
         ignorePath: true,
         ws: true,
+        xfwd: true,
         // we limit proxy requests to 30s by default, in development
         // we don't time out WebSocket requests to allow proxying
         proxyTimeout: proxyTimeout === null ? undefined : proxyTimeout || 30000,
@@ -29,6 +30,6 @@ export async function proxyRequest(req, res, parsedUrl, upgradeHead, reqBody, pr
     // object will detect the disconnect, and we can abort the proxy's
     // connection.
     proxy.on('proxyReq', (proxyReq)=>{
         res.on('close', ()=>proxyReq.destroy());
     });
     proxy.on('proxyRes', (proxyRes)=>{
diff --git a/dist/server/lib/router-utils/proxy-request.js b/dist/server/lib/router-utils/proxy-request.js
index de8549da03855f8ac804a26a240184d443dec6fc..09154b3edeea0fcacf135615fea3837c56aa9aff 100644
--- a/dist/server/lib/router-utils/proxy-request.js
+++ b/dist/server/lib/router-utils/proxy-request.js
@@ -1,4 +1,5 @@
 "use strict";
+const HttpProxy = require('next/dist/compiled/http-proxy');
 Object.defineProperty(exports, "__esModule", {
     value: true
 });
@@ -22,12 +23,12 @@ async function proxyRequest(req, res, parsedUrl, upgradeHead, reqBody, proxyTime
     delete parsedUrl.query;
     parsedUrl.search = (0, _serverrouteutils.stringifyQuery)(req, query);
     const target = _url.default.format(parsedUrl);
-    const HttpProxy = require('next/dist/compiled/http-proxy');
     const proxy = new HttpProxy({
         target,
         changeOrigin: true,
         ignorePath: true,
         ws: true,
+        xfwd: true,
         // we limit proxy requests to 30s by default, in development
         // we don't time out WebSocket requests to allow proxying
         proxyTimeout: proxyTimeout === null ? undefined : proxyTimeout || 30000,
@@ -44,6 +45,6 @@ async function proxyRequest(req, res, parsedUrl, upgradeHead, reqBody, proxyTime
     // object will detect the disconnect, and we can abort the proxy's
     // connection.
     proxy.on('proxyReq', (proxyReq)=>{
         res.on('close', ()=>proxyReq.destroy());
     });
     proxy.on('proxyRes', (proxyRes)=>{
